src/model/chord.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{Pitch, Quality};
+use combine::{Parser, Stream};
+use std::fmt;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Chord {
+    pub pitch: Pitch,
+    pub quality: Quality,
+}
+
+impl Chord {
+    pub fn new(pitch: Pitch, quality: Quality) -> Self {
+        Self { pitch, quality }
+    }
+
+    pub fn parser<Input>() -> impl Parser<Input, Output = Self>
```

Comment:
I don't really have a great understanding of this (I think this is the first time I've actually used the `impl Something` syntax, prompted by the parser docs), but I believe that it's actually preserving a concrete type there, just saving you from writing it out, rather than erasing the type by putting the value behind a pointer (the `Box` approach). It's less flexible than `Box`, because of that: you can't do `Vec<impl Parser>`, for example, because it does know the sizes of all `impl Parser`s, but they are not all the same, so can't be stored in a `Vec` like that.